etc/clang-format.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+. "$(dirname "${BASH_SOURCE[0]}")/../.evergreen/init.sh"
+
+: "${CLANG_FORMAT_VERSION:=15.0.7}"
+export PIPX_HOME="${BUILD_CACHE_DIR}/pipx"
+
+if ! run_python -c ''; then
+  fail "No Python found?"
+fi
+
+if ! run_python -c 'import pipx'; then
+  fail "No PipX found?"
+fi
+
+printf '' | run_python -m pipx run "clang-format==$CLANG_FORMAT_VERSION" "$@"
```

Comment:
Suggest documenting the purpose of the `printf ''` being piped to pipx.

Also suggest using `${CLANG_FORMAT_VERSION:?}`; if user-defined, it still shouldn't be null.

Reply:
I am observing the following error when running this script locally:
```
'clang-format==15.0.7' executable script not found in package 'clang-format==15.0.7'. Available executable scripts: clang-format, clang-format-diff.py, git-clang-format
```
Changing `clang-format==$CLANG_FORMAT_VERSION` to simply `clang-format` seems to resolve this issue, but I'm unsure this is the correct fix. Can/should the required clang-format version be specified by another means?

Reply:
That's odd. What version of pipx/Python are you using?

Reply:
Python 3.8.10 with pipx 0.16.4. If I edit `init.sh` to use Python 3.11 instead, it uses pipx 1.1.0 and I no longer observe the error.

Reply:
Ahh. Should we just require a new-enough pipx?

Reply:
I think so. [Release notes](https://pypa.github.io/pipx/changelog/#0170) suggest this was implemented in 0.17.0, which seems consistent with the observed failure with 0.16.4.